From 0a0007d594db2fceed413cd73e7f08dd8d8ddd57 Mon Sep 17 00:00:00 2001
From: Satish Kumar <satish.kumar01@arm.com>
Date: Sun, 13 Feb 2022 09:01:10 +0000
Subject: [PATCH] Fix: Crypto interface structure aligned with tf-m change.

NO NEED TO RAISE PR: The PR for this FIX  is raied by Emek.

Upstream-Status: Pending [Not submitted to upstream yet]
Signed-off-by: Emekcan Aras <Emekcan.Aras@arm.com>


---
 components/service/crypto/backend/psa_ipc/crypto_ipc_backend.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/components/service/crypto/backend/psa_ipc/crypto_ipc_backend.h b/components/service/crypto/backend/psa_ipc/crypto_ipc_backend.h
index c13c20e8..ec25eaf8 100644
--- a/components/service/crypto/backend/psa_ipc/crypto_ipc_backend.h
+++ b/components/service/crypto/backend/psa_ipc/crypto_ipc_backend.h
@@ -38,7 +38,8 @@ struct psa_ipc_crypto_pack_iovec {
 				      *   multipart operation
 				      */
 	uint32_t capacity;             /*!< Key derivation capacity */
-
+	uint32_t ad_length;            /*!< Additional Data length for multipart AEAD */
+	uint32_t plaintext_length;     /*!< Plaintext length for multipart AEAD */
 	struct psa_ipc_crypto_aead_pack_input aead_in; /*!< FixMe: Temporarily used for
 							    *   AEAD until the API is
 							    *   restructured
